### PR TITLE
fix(pii): Mark sentry_user field as PII

### LINKED ISSUE #3917: `sentry_user` not PII scrubbed
`sentry_user` added by Relay is not PII scrubbed if the user is PII scrubbed.

```json
"user":{
   "ip_address":null,
   "geo":{
      "country_code":"US",
      "city":"New York",
      "subdivision":"New York",
      "region":"United States"
   },
   "sentry_user":"ip:111.222.222.111"
},
```

The PII scrubbing metadata on the event:

```json
"_meta":{
   "user":{
      "ip_address":{
         "":{
            "rem":[
               [
                  "@ip:replace",
                  "s",
                  0,
                  4
               ],
               [
                  "@anything:remove",
                  "x"
               ]
            ],
            "len":15
         }
      }
   }
},

```

### PR DESCRIPTION
**Pull request by loewenheim**
This marks `User::sentry_user` as PII so it gets scrubbed correctly.

Fixes https://github.com/getsentry/relay/issues/3917.

#skip-changelog

### CODE REVIEW

**Review by Dav1dde — COMMENTED** [review]
Is extracting after PII scrubbing feasible?

This works now for IP addresses but not custom rules where the Sentry user specifically targets the ip field.

File: relay-event-schema/src/protocol/user.rs [hunk 1]
```diff
@@ -81,7 +81,7 @@ pub struct User {
     /// This field is computed by concatenating the name of specific fields of the `User`
     /// struct with their value. For example, if `id` is set, `sentry_user` will be equal to
     /// `"id:id-of-the-user".
-    #[metastructure(skip_serialization = "empty")]
+    #[metastructure(pii = "true", skip_serialization = "empty")]
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/user.rs:76]: mini nit: I would put this field declaration right after `name`, i.e. group it with the fields it is constructed from.

**Comment by loewenheim** [comment]
> Is extracting after PII scrubbing feasible?
> 
> This works now for IP addresses but not custom rules where the Sentry user specifically targets the ip field.

I tried switching around PII stripping and normalization (where `sentry_user` is set) and it broke a test by changing the reported browser name. I don't think I can predict the consequences of making that change.

**Comment by jjbayer** [comment]
> I tried switching around PII stripping and normalization (where `sentry_user` is set) and it broke a test by changing the reported browser name. I don't think I can predict the consequences of making that change.

Yes I'm afraid we cannot move all of event normalization after PII scrubbing. But we could introduce a new "post-pii normalization" step that for now only contains the creation of the `sentry_user` field. Tricky part is making sure the trimming processor still runs after that.

**Review by Dav1dde — APPROVED** [review]
Fixes the issue at hand for now, thinking about re-ordering processing is probably a not so easy/big task, think we should move forward with this. @jjbayer wdyt?

**Review by jjbayer — APPROVED** [review]
> Fixes the issue at hand for now, thinking about re-ordering processing is probably a not so easy/big task, think we should move forward with.

Yep

**Pull request merged**